File: squares/SquareDetector.java
/**
 * Utility class for detecting numbers that are perfect squares.
 * 
 * @author ubernostrum
 * @version 1.414
 * @since 1.414
 */
public class SquareDetector {
    // No instantiation for you.
    private SquareDetector() {}

    /**
     * Determines whether the given number is a perfect square.
     * @param n the number to check.
     */
    public static boolean isSquare(int n) {
        // Negative numbers aren't squares.
        if (n < 0) {
            return false;
        }

        // Zero is a square.
        if (n == 0) {
            return true;
        }

        long total = 0;
        int next = 1;

        while (total <= n) {
            total += next;
            if (total == n) {
                return true;
            }
            next += 2;
        }
        return false;
    }

    /**
     * When run with a numeric value as the first and only command-line argument,
     * outputs whether that value is a perfect square.
     */
    public static void main(String[] args) {
        int input;

        try {
            input = Integer.parseInt(args[0]);
        } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
            System.out.println("Please provide a numeric argument.");
            return;
        }

        if (isSquare(input)) {
            System.out.printf("%d is a square.\n", input);
        } else {
            System.out.printf("%d is not a square.\n", input);
        }
    }
}

File: squares/SquareDetector.cs
namespace SquareDetector;

internal static class SquareDetector
{
    private static IEnumerable<int> Count()
    {
        var current = 0;
        while (true)
            yield return current++;
    }

    private static IEnumerable<long> Accumulate(this IEnumerable<int> source)
    {
        long sum = 0;
        foreach (var i in source)
        {
            sum += i;
            yield return sum;
        }
    }

    /// <summary>
    /// Determine whether the given value is a perfect square.
    /// </summary>
    /// <param name="n">The value to check.</param>
    /// <returns>Whether the given value is a perfect square.</returns>
    private static bool IsSquare(int n) =>
        n >= 0 && (n == 0 || Count().Where(i => (i & 1) == 1).Accumulate().TakeWhile(s => s <= n).Contains(n));

    private static void Main(string[] args)
    {
        if (args.Length < 1 || !int.TryParse(args[0], out var input))
        {
            Console.WriteLine("Please provide a numeric argument.");
            return;
        }

        Console.WriteLine(IsSquare(input) ? $"{input} is a square." : $"{input} is not a square.");
    }
}